stdlib/LinearAlgebra/src/bidiag.jl

```diff
@@ -205,11 +205,13 @@ convert(T::Type{<:Bidiagonal}, m::AbstractMatrix) = m isa T ? m : T(m)
 similar(B::Bidiagonal, ::Type{T}) where {T} = Bidiagonal(similar(B.dv, T), similar(B.ev, T), B.uplo)
 similar(B::Bidiagonal, ::Type{T}, dims::Union{Dims{1},Dims{2}}) where {T} = zeros(T, dims...)
 
-function kron(A::Diagonal, B::Bidiagonal)
+# We can only handle Vector representations, otherwise _droplast! might not be defined
+function kron(A::Diagonal{<:Any,<:Vector}, B::Bidiagonal{<:Any,<:Vector})
     kdv = kron(diag(A), B.dv)
     kev = _droplast!(kron(diag(A), _pushzero(B.ev)))
```

Comment:
With your `_makevec` above, can't you just have

```suggestion
    kev = _droplast!(_makevec(kron(diag(A), _pushzero(B.ev))))
```

and avoid the extra method?

Reply:
That's a nice idea. I'll have to call `_makevec` for both `kdv` and `kev` because they need to have the same type, but I like it anyway.

Reply:
Done.

Reply:
I think the "homogenization" of the two vectors to same type will be done within the constructor anyway, so it wouldn't be strictly necessary. OTOH, this performs the same amount of work and allocations, so should be fine.

Reply:
No, this does not happen:
```Julia
julia> Bidiagonal([1,2], sparsevec([3]), :U)
ERROR: MethodError: no method matching Bidiagonal(::Vector{Int64}, ::SparseVector{Int64, Int64}, ::Symbol)
```
(although maybe it should happen...)